proto/private/pbstorage/raft.proto

```diff
@@ -1,25 +1,103 @@
 syntax = "proto3";
 
+// This package contains types used by the Raft storage backend that lives in
+// the internal/storage/raft Go package.
 package hashicorp.consul.internal.storage.raft;
 
+import "annotations/ratelimit/ratelimit.proto";
+import "google/protobuf/empty.proto";
 import "pbresource/resource.proto";
 
-message Request {
-  enum Op {
-    OP_UNSPECIFIED = 0;
-    OP_WRITE = 1;
-    OP_DELETE = 2;
+// Forwarding service is used for forwarding write and consistent read
+// operations to the Raft leader. It is served on Consul's multiplexed
+// server port, which is the same port used for regular Raft traffic.
+service Forwarding {
+  // RoundTrip handles the forwarded operation.
+  rpc RoundTrip(Request) returns (Response) {
```

Comment:
What are the tradeoffs of the single `RoundTrip` RPC approach vs something like:

```
service Forwarding {
   rpc ForwardRead(ReadRequest) returns (ReadResponse) ;
   rpc ForwardList(ListRequest) returns (ListResponse) ;
   rpc ForwardWrite(WriteRequest) returns (WriteResponse) ;
   rpc ForwardDelete(DeleteRequest) returns (DeleteResponse) ;
}
```

I can see that the expanded RPC form will have more methods to implement, however it looks like in internal/storage/raft/forwarding.go you ended up with a big switch statement and all the individual methods anyways. 

The big advantage I see to the expanded form is that there is no implicit contract regarding sending one type of request and getting back a specific type of response. With the expanded RPC approach the valid request/response type are a part of the RPC definition and the contents have no variability to deal with and ensure the consistency of within the code.

Reply:
I'd originally gone with the single endpoint for symmetry with the `Backend.raftApply` method (previously called `roundTrip`) which needs a wrapping message and type enum for the Raft log entries.

I think you're right that the expanded form makes more sense for forwarding, though, as without it we're sort of implementing an RPC system on top of an RPC system 😅